src/file.c

```diff
@@ -531,18 +536,31 @@ size_t RETRACE_IMPLEMENTATION(fwrite)(const void *ptr, size_t size, size_t nmemb
 	event_info.return_value_type = PARAMETER_TYPE_INT;
 	event_info.return_value = &r;
 	event_info.logging_level = RTR_LOG_LEVEL_NOR;
-	retrace_log_and_redirect_before(&event_info);
 
-	if (rtr_str_inject(STRINJECT_FUNC_FWRITE, (void *) ptr, size * nmemb)) {
+	if (size == 1 && rtr_str_inject(STRINJECT_FUNC_FWRITE, (void *) ptr, size * nmemb, &inject_buffer, &inject_len)) {
```

Comment:
`size == 1` so `size * nmemb == nmemb`

Reply:
yes, right

Reply:
also `fwrite("hello", 5, 1, f)` is probably the same as `fwrite("hello", 1, 5, f)` but only one will get injected. Is that intentional?

Reply:
yes, not possible to inject using fwrite when size is more than 1
but other functions are possible